lib/App/Netdisco/Worker/Plugin/Macsuck/PortAccessEntity.pm

```diff
@@ -0,0 +1,72 @@
+package App::Netdisco::Worker::Plugin::Macsuck::PortAccessEntity;
+
+use Dancer ':syntax';
+use App::Netdisco::Worker::Plugin;
+use aliased 'App::Netdisco::Worker::Status';
+use Data::Dumper;
+use Dancer::Plugin::DBIC 'schema';
+
+use App::Netdisco::Util::Worker;
+use App::Netdisco::Util::Permission qw/check_acl_no check_acl_only/;
+
+register_worker({ phase => 'early', driver => 'snmp' }, sub {
+
+  no warnings "uninitialized";
+  my ($job, $workerconf) = @_;
+  my $count = 0;
+  my $device = $job->device;
+  my $snmp = App::Netdisco::Transport::SNMP->reader_for($device)
+    or return Status->defer("pae failed: could not SNMP connect to $device");
+  my $interfaces = $snmp->interfaces;
+  debug sprintf ' [%s] pae - updating PortAccessEntity details for %s', $device->ip, $device->dns;
+
+  # device property
+  my $pae_control = $snmp->pae_control();
+
+  if ($pae_control) {
+        schema('netdisco')->resultset('Device')->search({ 'me.ip' => $device->ip})
+          ->update({ pae_control => $pae_control });
+    debug sprintf ' [%s] pae - PortAccessEntity device-wide support: %s', $device->ip, $pae_control;
+  } else {
+    debug sprintf ' [%s] pae - no PortAccessEntity support, leaving worker', $device->ip;
+    return Status->info("Skipped pae for $device");
+  }
+
+  # individual port properties
+  my $pae_authconfig_state = $snmp->pae_authconfig_state();
+  my $pae_authconfig_port_status = $snmp->pae_authconfig_port_status();
+  my $pae_authsess_user = $snmp->pae_authsess_user();
+  my $pae_authsess_mab = $snmp->pae_authsess_mab();
+  my $pae_capabilities = $snmp->pae_i_capabilities();
+  my $capdesc = { 'dot1xPaePortAuthCapable' => , 'Authenticator functions are supported',
+   'dot1xPaePortSuppCapable' => , 'Supplicant functions are supported'};
```

Comment:
So for LLDP we don't store the capabilities and instead have boolean fields for the capabilities. Would you consider doing the same here? (such as `pae_authenticator_capable` and `pae_supplicant_capable`)

I'm not so happy for long english text to be stored in the field, and that would be better done in the user interface in response to the setting in the database.

Also can a port have multiple PAE capabilities? Because the current code does not support it.

Reply:
Yes right it's actually a bit field, not an enum, so both could be true. Should work in the new commit.